### DISCUSSION TITLE
Deal with complex data with use Mutation?

### DISCUSSION THREAD

**Discussion by codernirmalnp** [TanStack/query#2093] [Q&A]
![api response](https://user-images.githubusercontent.com/61540469/113719521-6a605800-970d-11eb-8bf0-61cd4b6874e9.JPG)
I have data stored as global in react-query. I have fetched it using useQuery.I want to Updates from Mutation Responses which 
will render in brower after submittion.I render category part of store in browser.
i want such like this and update my browser data
const mutation = useMutation(editTodo, {
   onSuccess: data => queryClient.setQueryData(['todo', { id: 5 }], data),
 })
 How can i Acheive like this/

**Comment by TkDodo** [comment]
sorry, I don't understand the question. you can update the queryCache as you like with `setQueryData` in the `onSuccess` callback of `useMutation` - if your backend returns the correct structure, you can set it like you've shown. Can you maybe describe what is happening, what is not working and what you would expect instead? A running codesandbox example also always helps.
